packages/material-ui/src/Select/SelectInput.js

```diff
@@ -14,6 +14,10 @@ function areEqualValues(a, b) {
   return String(a) === String(b);
 }
 
+function isEmptyString(val) {
+  return typeof val === 'string' && !val.match(/\S/);
```

Comment:
```suggestion
  return typeof val === 'string' && !val.trim().length;
```
Don't know if this is faster or slower but it probably minimizes better (not tested)

Reply:
Would it work?
```suggestion
return typeof val === 'string' && val.trim();
```

Might be time to add a small unit test for it :)

Reply:
Add the `!` back and yeah, it would